pkg/common/utils/gang_utils.go

```diff
@@ -75,7 +75,7 @@ func GetPlaceholderResourceRequest(resources map[string]resource.Quantity) v1.Re
 }
 
 func GetPlaceholderFlagFromPodSpec(pod *v1.Pod) bool {
-	if value, ok := pod.Annotations[constants.AnnotationPlaceholderFlag]; ok {
+	if value, ok := pod.Annotations[constants.PlaceholderFlag]; ok {
```

Comment:
Do we add the label just as a convenience or is there more behind this?
The appID retrieval uses a logic to look at annotations and then labels (normal followed by spark).

Reply:
I think we can follow the same logic in `GetApplicationIDFromPod()`, add the label part in `GetPlaceholderFlagFromPodSpec()`